torch_geometric/explain/algorithm/pg_explainer.py

```diff
@@ -0,0 +1,276 @@
+    uses a neural network :obj:`explainer_model` , to predict which
+    edges are crucial to a GNNs node or graph prediction.
+
+    Args:
+        epochs (int, optional): The number of epochs to train.
+            (default: :obj:`30`).
+        lr (float, optional): The learning rate to apply.
+            (default: :obj:`0.003`).
+        log (bool, optional): If set to :obj:`False`, will not log any learning
+            progress. (default: :obj:`True`)
+        **kwargs (optional): Additional hyper-parameters to override default
+            settings in
+            :attr:`~torch_geometric.explain.algorithm.PGExplainer.coeffs`.
+    """
+
+    coeffs = {
+        'edge_size': 0.05,
+        'edge_ent': 1.0,
+        'temp': [5.0, 2.0],
+        'bias': 0
+    }
+
+    def __init__(self, epochs: int = 30, lr: float = 0.003, log: bool = True,
+                 **kwargs):
+        super().__init__(training_need=True)
+        self.epochs = epochs
+        self.lr = lr
+        self.log = log
+        self.coeffs.update(kwargs)
+
+    def supports(self) -> bool:
+        task_level = self.model_config.task_level
+        if task_level not in [ModelTaskLevel.node, ModelTaskLevel.graph]:
+            logging.error(f"Task level '{task_level.value}' not supported")
+            return False
+
+        edge_mask_type = self.explainer_config.edge_mask_type
+        if edge_mask_type not in [MaskType.object]:
+            logging.error(f"Edge mask type '{edge_mask_type.value}' not "
+                          f"supported")
+            return False
+
+        node_mask_type = self.explainer_config.node_mask_type
+        if node_mask_type is not None:
+            logging.error(
+                "Node mask not supported. Set 'node_mask_type' to 'None'")
+            return False
+
+        is_hetero = self._is_hetero
+        if is_hetero:
+            logging.error("Heterogeneous graphs not supported.")
+            return False
+
+        return True
+
+    def train_explainer(self, model: torch.nn.Module, x: Tensor,
+                        edge_index: Tensor, target: Tensor = None,
+                        index: Tensor = None, **kwargs):
+        # Initialize trainable parameters.
+        z = get_intermediate_messagepassing_embeddings(model, x=x,
+                                                       edge_index=edge_index,
+                                                       **kwargs)
+        out_channels = z.shape[-1]
+        task_level = self.model_config.task_level
+        self.exp_in_channels = 2 * out_channels if (
+            task_level == ModelTaskLevel.graph) else 3 * out_channels
+        self.explainer_model = nn.Sequential(
+            nn.Linear(self.exp_in_channels, 64),
+            nn.ReLU(),
+            nn.Linear(64, 1),
+        )
+
+        assert x.shape[0] == z.shape[0]
+        batch = kwargs.get('batch', None)
+        assert ~(batch is None and index is None)
```

Comment:
I'm not sure if I understand why we always need a `batch`: what happens if I want to explain a single graph for a graph regression problem ? 

If we always require a `batch`, should we add it as an explicit parameter ?

Reply:
Good point. Removed requirement for `batch`.